src/Illuminate/Cache/ClassUsesRecursive.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Illuminate\Cache;
+
+class ClassUsesRecursive
+{
+    private static array $classes = [];
+
+    /**
+     * @param $className
+     * @return array
+     */
+    public static function classUsesRecursive($className): array
+    {
+        if (! isset(self::$classes[$className])) {
+            self::$classes[$className] = class_uses_recursive($className);
+        }
+
+        return self::$classes[$className];
+    }
+
+    /**
+     * @param $needleClassName
+     * @param $haystackClassName
+     * @return bool
+     */
+    public static function inArray($needleClassName, $haystackClassName): bool
```

Comment:
How do you feel about naming it `uses()`?

Reply:
I'm totally aware about the naming.